src/PrestaShopBundle/Command/GenerateMailTemplatesCommand.php

```diff
@@ -0,0 +1,108 @@
+<?php
+/**
+ * 2007-2018 PrestaShop.
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to http://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2018 PrestaShop SA
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShopBundle\Command;
+
+use PrestaShop\PrestaShop\Core\Exception\InvalidException;
+use PrestaShop\PrestaShop\Core\MailTemplate\MailTemplateGenerator;
+use Symfony\Bundle\FrameworkBundle\Command\ContainerAwareCommand;
+use Symfony\Component\Console\Input\InputArgument;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+use Language;
+
+class GenerateMailTemplatesCommand extends ContainerAwareCommand
+{
+    protected function configure()
+    {
+        $this
+            ->setName('prestashop:mail:generate')
+            ->setDescription('Generate mail templates for a specified theme')
+            ->addArgument('theme', InputArgument::REQUIRED, 'Theme to use for mail templates.')
+            ->addArgument('locale', InputArgument::REQUIRED, 'Which locale to use for the templates.')
+            ->addArgument('outputFolder', InputArgument::REQUIRED, 'Output folder to export templates.')
+        ;
+    }
+
+    /**
+     * @param InputInterface $input
+     * @param OutputInterface $output
+     *
+     * @return int|void|null
+     */
+    protected function execute(InputInterface $input, OutputInterface $output)
+    {
+        $theme = $input->getArgument('theme');
+        $outputFolder = $input->getArgument('outputFolder');
+        if (!file_exists($outputFolder)) {
+            $outputFolder = implode(DIRECTORY_SEPARATOR, [getcwd(), $outputFolder]);
```

Comment:
Don't use getcwd(), it depends on which directory you are running the command. Use `kernel.root_dir` instead?

```php
var_dump(getcwd());
```
```bash
php test.php 
/home/got/test.php:2:
string(9) "/home/got"
[3.6.3] got@debian:~$ cd projects/
[3.6.3] got@debian:~/projects$ php ../test.php 
/home/got/test.php:2:
string(18) "/home/got/projects"
```

Reply:
I know that's the purpose, so that you could execute the command this way:

```bash
mkdir test-folder
~/MyPrestaShop/bin/console prestashop:mail:generate classic en test-folder
```

The purpose is to compute the relatif path, so it starts from the executing folder

Reply:
If you're running this script in a cron or anything else, you don't have executing folder :/
use realpath instead